src/python/pants/rules/core/test.py

```diff
@@ -79,9 +79,19 @@ class TestOptions(GoalSubsystem):
   @classmethod
   def register_options(cls, register) -> None:
     super().register_options(register)
-    register('--debug', type=bool, default=False,
-             help='Run a single test target in an interactive process. This is '
-                  'necessary, for example, when you add breakpoints in your code.')
+    register(
+      '--debug',
+      type=bool,
+      default=False,
+      help='Run a single test target in an interactive process. This is necessary, for example, when you add '
+           'breakpoints in your code.'
+    )
+    register(
+      '--run-coverage',
+      type=bool,
+      default=False,
+      help='Generate a coverage report for this test run.',
+    )
```

Comment:
This really should be named `--coverage` but it shades `--junit-coverage` (which is fixable) and `--pytest-coverage` (which expects to take arguments and will need a deprecation cycle.)